public/src/main/java/com/revenuecat/purchases/models/GoogleStoreProduct.kt

```diff
@@ -15,11 +14,41 @@ data class GoogleStoreProduct(
     override val title: String,
     override val description: String,
     override val subscriptionPeriod: String?,
-    override val purchaseOptions: List<GooglePurchaseOption>,
-    override val defaultOption: PurchaseOption?,
-    val productDetails: @RawValue ProductDetails // TODO parcelize?
+    override val purchaseOptions: List<PurchaseOption>,
+    override val defaultOption: PurchaseOption?
 ) : StoreProduct, Parcelable {
 
+    // @IgnoredOnParcel isn't allowed in primary constructor without default value
```

Comment:
i'm kinda stuck on what to do here. @vegaro @tonidero any ideas?

@RawValue doesn't work (cannot be marshalled error), there's no way to create a ProductDetails ourselves because no public constructor. and then see the comment for what else i've tried.

seems like if we really want this class to be parcelable, we might need to make ProductDetails nullable (or come up with some other fancy solution better than this hacky one i did). nullable is fine i guess, but annoying in other code, and also isn't really representative of a GoogleStoreProduct since it should never be null

Reply:
first solution [here](https://stackoverflow.com/questions/61758963/how-to-ignore-fields-when-using-parcelize-annotation-in-kotlin) doesn't work either, i get an error that Parcelable constructor parameter should be val or var.

maybe that second solution is possible

Reply:
Hmm right, this is tricky... Not sure if we have a contact at Google we could ask whether they would consider making the `ProductDetails` class parcelable/serializable or making the constructor public. Unlikely to do so though 😞 

Main problem I'm seeing is that since `ProductDetails` is not parcelable/serializable and constructor is not public, there is really no way to parcelize this object... The solution you have below would use the primary constructor of `GoogleStoreProduct` when trying to parcelize this class, so if it's parcelized, `ProductDetails` would be null...

As for making it nullable, since `ProductDetails` is needed to start purchases, we would need a way to get it in case it's null, but at that point, I think I would just remove it from this data class and access `ProductDetails` in a different way. I'm thinking storing the `ProductDetails` in an application-level cache, which would obviously be a pretty big change...

My first question, however, would be whether this class needs to be `Parcelable` or not... It would be very nice to be able to make it `Parcelable`, but with the way Google made `ProductDetails`, unless we really need to, I think we should make it non-parcelable.

Reply:
right, i'm thinking we just make it non-parcelable. we can see if we get a lot of pushback on it.

Reply:
@vegaro any thoughts?

Reply:
i'm going to reverse this commit for now, and create a separate ticket to follow-up. would still love feedback though

Reply:
As I mentioned in the other conversation, not having a public constructor really leaves us without many options here